app/controllers/users_controller.rb

```diff
@@ -21,10 +28,46 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.update_attributes(user_params)
+      flash[:sucess] = t ".profile_update"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
+    flash[:success] = t ".delete_user"
+    redirect_to users_path
+  end
+
   private
 
   def user_params
     params.require(:user).permit(:name, :email, :password,
       :password_confirmation)
   end
+
+  def logged_in_user
+    return if logged_in?
+    store_location
+    flash[:danger] = t "users.logged_in_user.please_log_in"
+    redirect_to login_path
+  end
+
+  def correct_user
+    @user = User.find_by id: params[:id]
```

Comment:
check @user có nil không đi em

Reply:
Oh I fixed it but it's not appear here, can you see it in "files changed" item, tks:))